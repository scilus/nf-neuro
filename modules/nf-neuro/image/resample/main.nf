process IMAGE_RESAMPLE {
    tag "$meta.id"
    label 'process_single'
    label 'process_high_memory'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
        'scilus/scilus:2.0.2' }"

    input:
    tuple val(meta), path(image), path(ref) /* optional, input = [] */

    output:
    tuple val(meta), path("*_resampled.nii.gz") , emit: image
    path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def suffix = task.ext.suffix ? "${task.ext.first_suffix}_resampled" : "resampled"
    def reference = "$ref" ? "--ref $ref" : ""
    def voxel_size = task.ext.voxel_size ? "--voxel_size " + task.ext.voxel_size : ""
    def volume_size = task.ext.volume_size ? "--volume_size " + task.ext.volume_size : ""
    def iso_min = task.ext.iso_min ? "--iso_min" : ""
    def interp = task.ext.interp ? "--interp " + task.ext.interp : ""
    def f = task.ext.f ? "-f" : ""
    def enforce_dimensions = task.ext.enforce_dimensions ? "--enforce_dimensions" : ""

    """
    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
    export OMP_NUM_THREADS=1
    export OPENBLAS_NUM_THREADS=1

    scil_volume_resample.py $image ${prefix}_${suffix}.nii.gz \
        $voxel_size $volume_size $reference $iso_min \
        $f $enforce_dimensions $interp

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(pip list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def suffix = task.ext.suffix ? "${task.ext.first_suffix}_resampled" : "resampled"
    """
    scil_volume_resample.py -h

    touch ${prefix}_${suffix}.nii.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(pip list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
