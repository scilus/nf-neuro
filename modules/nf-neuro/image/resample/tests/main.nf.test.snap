{
    "image - resample - isomin": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_resampled.nii.gz:md5,f3164907d16ceabf606a03ea0efd3a2f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,06181e25531a3ebdefdfd2e641a3d645"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_resampled.nii.gz:md5,f3164907d16ceabf606a03ea0efd3a2f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,06181e25531a3ebdefdfd2e641a3d645"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-24T15:30:19.061192"
    },
    "image - resample - nn": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_resampled.nii.gz:md5,8d3e1c9552673b1ce60bc4d090198564"
                    ]
                ],
                "1": [
                    "versions.yml:md5,06181e25531a3ebdefdfd2e641a3d645"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_resampled.nii.gz:md5,8d3e1c9552673b1ce60bc4d090198564"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,06181e25531a3ebdefdfd2e641a3d645"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-24T15:30:42.725713"
    },
    "image - resample - volsize": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_resampled.nii.gz:md5,7c83b666f9fb2d68643a20d1c7ac21d8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,06181e25531a3ebdefdfd2e641a3d645"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_resampled.nii.gz:md5,7c83b666f9fb2d68643a20d1c7ac21d8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,06181e25531a3ebdefdfd2e641a3d645"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-24T15:30:07.1626"
    },
    "image - resample - voxsize": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_resampled.nii.gz:md5,6ffdabab138186db2dab6f98ea6e9ffb"
                    ]
                ],
                "1": [
                    "versions.yml:md5,06181e25531a3ebdefdfd2e641a3d645"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_resampled.nii.gz:md5,6ffdabab138186db2dab6f98ea6e9ffb"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,06181e25531a3ebdefdfd2e641a3d645"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-24T15:29:55.01735"
    },
    "image - resample - ref": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_resampled.nii.gz:md5,0a45d21ff294fda2e41c620cf83d0406"
                    ]
                ],
                "1": [
                    "versions.yml:md5,06181e25531a3ebdefdfd2e641a3d645"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_resampled.nii.gz:md5,0a45d21ff294fda2e41c620cf83d0406"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,06181e25531a3ebdefdfd2e641a3d645"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-24T15:30:31.018615"
    }
}